aptible/log_drain.go

```diff
@@ -0,0 +1,135 @@
+package aptible
+
+import (
+	"fmt"
+
+	"github.com/aptible/go-deploy/client/operations"
+	"github.com/aptible/go-deploy/models"
+	strfmt "github.com/go-openapi/strfmt"
+)
+
+type LogDrain struct {
+	Deleted                bool
+	ID                     int64
+	Handle                 string
+	DrainType              string
+	DrainApps              bool
+	DrainDatabases         bool
+	DrainEphemeralSessions bool
+	DrainProxies           bool
+	DrainHost              string
+	DrainPort              int64
+	LoggingToken           string
+	URL                    string
+}
+
+type LogDrainCreateAttrs struct {
+	Database               strfmt.URI
+	DatabaseID             int64
+	DrainApps              bool
+	DrainDatabases         bool
+	DrainEphemeralSessions bool
+	DrainProxies           bool
+	DrainHost              strfmt.URI
+	DrainPassword          string
+	DrainPort              int64
+	DrainType              *string
+	LoggingToken           string
+	URL                    strfmt.URI
+}
+
+func (c *Client) CreateLogDrain(handle string, accountID int64, attrs *LogDrainCreateAttrs) (*LogDrain, error) {
+	request := &models.AppRequest16{
+		Database:               attrs.Database,
+		DatabaseID:             attrs.DatabaseID,
+		DrainApps:              attrs.DrainApps,
+		DrainDatabases:         attrs.DrainDatabases,
+		DrainEphemeralSessions: attrs.DrainEphemeralSessions,
+		DrainHost:              attrs.DrainHost,
+		DrainPassword:          attrs.DrainPassword,
+		DrainPort:              attrs.DrainPort,
+		DrainType:              attrs.DrainType,
+		Handle:                 &handle,
+		LoggingToken:           attrs.LoggingToken,
+		URL:                    attrs.URL,
+	}
+	logDrain := &LogDrain{}
+	params := operations.NewPostAccountsAccountIDLogDrainsParams().WithAccountID(accountID).WithAppRequest(request)
+	response, err := c.Client.Operations.PostAccountsAccountIDLogDrains(params, c.Token)
+	if err != nil {
+		return logDrain, err
+	}
+
+	if response.Payload.ID == nil {
```

Comment:
You need to call the `provision` operation on the log drain so that it's actually created. This only makes an API record but there won't be a real log drain until the provision is executed.

Pretty everything follows that pattern but here's an example: https://github.com/aptible/go-deploy/blob/810f7fb981bd26f9e197d9bf973f0dab3ae3968f/aptible/endpoints.go#L76